### PR TITLE
[#5602] feat(core): Add storage schema for model (part-1)

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

This PR adds the 1st part of storage schema for model metadata.

### Why are the changes needed?

Fix: #5602 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Add various UTs to cover the code.

### CODE REVIEW

File: scripts/h2/schema-0.8.0-h2.sql [hunk 1]
```diff
@@ -289,3 +289,51 @@ CREATE TABLE IF NOT EXISTS `owner_meta` (
...
+
+CREATE TABLE IF NOT EXISTS `model_version_alias_rel` (
+    `id` BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'auto increment id',
+    `model_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'model id',
+    `model_version` INT UNSIGNED NOT NULL COMMENT 'model version',
+    `model_version_alias` VARCHAR(128) NOT NULL COMMENT 'model version alias',
+    `deleted_at` BIGINT(20) UNSIGNED NOT NULL DEFAULT 0 COMMENT 'model version alias deleted at',
+    PRIMARY KEY (`id`),
+    UNIQUE KEY `uk_mi_mv_mva_del` (`model_id`, `model_version`, `model_version_alias`, `deleted_at`),
+    KEY `idx_mvtid` (`model_id`),
```
Comment (yuqi1129) [inline scripts/h2/schema-0.8.0-h2.sql:337]: I believe the unique key `uk_mi_mv_mva_del` can cover key `idx_mvtid` according to the left match rule.

File: scripts/h2/schema-0.8.0-h2.sql [hunk 2]
```diff
@@ -289,3 +289,51 @@ CREATE TABLE IF NOT EXISTS `owner_meta` (
...
+CREATE TABLE IF NOT EXISTS `model_version_info` (
+    `id` BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'auto increment id',
+    `metalake_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'metalake id',
+    `catalog_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'catalog id',
+    `schema_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'schema id',
+    `model_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'model id',
+    `version` INT UNSIGNED NOT NULL COMMENT 'model version',
+    `model_version_comment` TEXT DEFAULT NULL COMMENT 'model version comment',
+    `model_version_properties` MEDIUMTEXT DEFAULT NULL COMMENT 'model version properties',
+    `model_version_uri` MEDIUMTEXT NOT NULL COMMENT 'model storage uri',
```
Comment (yuqi1129) [inline scripts/h2/schema-0.8.0-h2.sql:319]: Is it possible that the `model_version_uri` is too large to use `MEDIUMTEXT`?

Comment (jerryshao) [reply scripts/h2/schema-0.8.0-h2.sql:319]: Yeah, looks so, I will change it.

File: scripts/h2/schema-0.8.0-h2.sql (hunk 1 above)
Comment (xloya) [reply scripts/h2/schema-0.8.0-h2.sql:337]: Yeah, I agree with this.

**Comment by jerryshao** [comment]
@yuqi1129 @xloya  would you please help to review again?

**Comment by xloya** [comment]
I have no more question, LGTM.

File: core/src/main/java/org/apache/gravitino/storage/relational/service/ModelMetaService.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+import org.apache.gravitino.storage.relational.utils.ExceptionUtils;
+import org.apache.gravitino.storage.relational.utils.POConverters;
+import org.apache.gravitino.storage.relational.utils.SessionUtils;
+import org.apache.gravitino.utils.NameIdentifierUtil;
+import org.apache.gravitino.utils.NamespaceUtil;
+import org.slf4j.Logger;
+
+public class ModelMetaService {
+
+  private static final Logger LOG = org.slf4j.LoggerFactory.getLogger(ModelMetaService.class);
```
Comment (yuqi1129) [inline core/src/main/java/org/apache/gravitino/storage/relational/service/ModelMetaService.java:43]: Better to simplify the class path here.

**Review by yuqi1129 — COMMENTED** [review]
Just a minor one, others looks good to me.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**